### PR TITLE
Add org.freedesktop.Sdk.Extension.texlive

### PR DESCRIPTION
**Pull request by Alexander-Wilms**

### CODE REVIEW

**Comment by Alexander-Wilms** [comment]
The install script still takes a long time to run, but I think it's faster than installing TeX Live from the .iso

**Comment by nedrichards** [comment]
bot, build org.freedesktop.Sdk.Extension.texlive

**Comment by barthalion** [comment]
Failed to build on aarch64, if you haven't noticed.

**Comment by Alexander-Wilms** [comment]
I've added a flathub.json file. @wolfv and @somas95 , would you be interested in using this for UberWriter? I think it should be possible to enable TeX Live at runtime by adding /usr/lib/sdk/texlive/bin to PATH.

**Comment by wolfv** [comment]
Cool!

**Comment by nedrichards** [comment]
bot, build org.freedesktop.Sdk.Extension.texlive

**Comment by somas95** [comment]
I'll look into this! It's the whole package?

**Comment by Alexander-Wilms** [comment]
Yes, it's the entirety of TeX Live, 3.1 GB.

**Comment by Alexander-Wilms** [comment]
Maybe sticking with `flatpak-spawn --host` and the TeX Live version provided by the distro is a better choice...

**Comment by JakobJakobson** [comment]
> Maybe sticking with flatpak-spawn --host and the TeX Live version provided by the distro is a better choice...
>
> I was able to build a fairly complex document using lualatex, biber and makeglossaries just fine this way.

Hi. I'm sorry to barge in at the late stage of discussion. But isn't the idea of the flatpak packages that you would be able to use latex by installing the flatpak package even though it is not installed on the system by the distro-specific installer as it is either not available for the linux distribution or it is not installed by the admin? So installing 3.1 GB takes much space but in the end it is the only way if you don't want or can't rely on the distro-specific installation. CentOS 7 for examples only has a partial texlive 2012 in its repositories and there is little gain using that over a current texlive.

The sum it up: Personally, I would favor such an extensions as there are multiple latex editors out there that could share that extension.

**Comment by Alexander-Wilms** [comment]
@nedrichards Could this be merged?

**Comment by nedrichards** [comment]
@Alexander-Wilms What's the plan for the 18.08 FDo runtime?

**Comment by Alexander-Wilms** [comment]
Could I create two branches, like how it's done for QtWebkit: https://github.com/flathub/io.qt.qtwebkit.BaseApp/tree/5.11 ?

**Comment by barthalion** [comment]
Yes, we can; I will merge this one as '1.6' branch. Let's just make sure it still builds.

**Comment by barthalion** [comment]
bot, build org.freedesktop.Sdk.Extension.texlive

File: org.freedesktop.Sdk.Extension.texlive.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  - cp *.sh /usr/lib/sdk/texlive/
+  sources:
+  - type: script
+    commands:
+    - install -d /app
+    - cp -ra /usr/lib/sdk/texlive/* /app/
+    dest-filename: install.sh
+  - type: script
+    commands:
+    - export PATH=/usr/lib/sdk/texlive/bin/x86_64-linux:/usr/lib/sdk/texlive/bin:$PATH
```
Comment (nedrichards) [inline org.freedesktop.Sdk.Extension.texlive.yaml]: this script isn't marked x86-64 only. Is it going to work on arm and i386 as well?

Comment (Alexander-Wilms) [reply org.freedesktop.Sdk.Extension.texlive.yaml]: Probably not. I changed the flathub.json file.

**Comment by nedrichards** [comment]
bot, build org.freedesktop.Sdk.Extension.texlive

**Comment by Alexander-Wilms** [comment]
@nedrichards Could you merge this please? That would make building the RStudio Flatpak easier.

File: org.freedesktop.Sdk.Extension.texlive.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
+id: org.freedesktop.Sdk.Extension.texlive
+branch: 1.6
+runtime: org.freedesktop.Sdk
+build-extension: true
+sdk: org.freedesktop.Sdk
+runtime-version: 1.6
```
Comment (barthalion) [inline org.freedesktop.Sdk.Extension.texlive.yaml]: Can we build it for 18.08? 1.6 is on its way out.

**Comment by barthalion** [comment]
bot, build org.freedesktop.Sdk.Extension.texlive

**Comment by cvfosammmm** [comment]
I'm looking forward to using this with my app, see https://gitlab.gnome.org/cvfosammmm/setzer/issues/3

**Comment by Alexander-Wilms** [comment]
bot, build org.freedesktop.Sdk.Extension.texlive

**Comment by Alexander-Wilms** [comment]
testbot, build org.freedesktop.Sdk.Extension.texlive

**Comment by barthalion** [comment]
bot, build org.freedesktop.Sdk.Extension.texlive

**Comment by Alexander-Wilms** [comment]
Succeeded: https://flathub.org/builds/#/builders/2/builds/7318

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/org.freedesktop.Sdk.Extension.texlive

Thanks!

**Pull request closed without merging**